### ISSUE TITLE
Trailing slash not checked?

### ISSUE THREAD

**Issue by rapodaca** [thruster-rs/Thruster#196]
An example taken from the examples directory:

```rust
use hyper::Body;
use log::info;
use thruster::context::basic_hyper_context::{
    generate_context, BasicHyperContext as Ctx, HyperRequest,
};
use thruster::hyper_server::HyperServer;
use thruster::{m, middleware_fn};
use thruster::{App, ThrusterServer};
use thruster::{MiddlewareNext, MiddlewareResult};

#[middleware_fn]
async fn plaintext(mut context: Ctx, _next: MiddlewareNext<Ctx>) -> MiddlewareResult<Ctx> {
    let val = "Hello, World!";
    context.body = Body::from(val);
    Ok(context)
}

fn main() {
    env_logger::init();
    info!("Starting server...");

    let mut app = App::<HyperRequest, Ctx, ()>::create(generate_context, ());
    app.get("/plaintext", m![plaintext]);

    let server = HyperServer::new(app);
    server.start("0.0.0.0", 4321);
}
```

I expect `/plaintext` to match and it does. But `/plaintext/` also matches, which is unexpected.

If I change the route to `/plaintext/`, then '/plaintext` matches, which I also didn't expect. I looked for this in documentation, an even source, but couldn't find anything.

Is the trailing slash matching part of Thruster, or one of the underlying libraries? What should be the supported behavior?

**Comment by trezm** [comment]
That's correct! We model somewhat after express here, in that `/route/ == /route` in terms of matching. This is something I could update though, is there a use case you're after?

**Comment by rapodaca** [comment]
I'm not sure the current behavior is correct. See these two items:

- [OK to skip slash before query string?](https://stackoverflow.com/questions/13442377/redirect-all-trailing-slashes-globally-in-express/43948185)
- [Why trailing slashes on URIs are important](https://cdivilly.wordpress.com/2014/03/11/why-trailing-slashes-on-uris-are-important/)

Bottom line: treating `/foo/` and `/foo` as synonyms for the same resource can interfere with the use of relative URLs. A redirect, which can be applied as a top-level middleware by those users who want it, solves the problem without introducing another. At least that's my understanding.

**Comment by trezm** [comment]
I'm not sure I fully agree with some of the points in that article, but I
do fully agree that it should be the developer's choice. I'll work on
adding a configuration for it! Thanks for raising the issue :-)

**Comment by rapodaca** [comment]
It looks like this has been resolved in v1.1.10. Looking forward to trying it out!
